TopNews/app/src/main/java/com/fisfam/topnews/fragment/TopicFragment.java

```diff
@@ -9,14 +9,46 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.fisfam.topnews.R;
+import com.fisfam.topnews.adapter.TopicAdapter;
+import com.fisfam.topnews.pojo.Category;
+
+import java.util.ArrayList;
 
 public class TopicFragment extends Fragment {
 
+    private View mRootView;
+    private RecyclerView mRecyclerView;
+    private RecyclerView.Adapter mAdapter;
+    private RecyclerView.LayoutManager mLayoutManager;
+
     @Nullable
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
-        Toast.makeText(getContext(), "TopicFragment launched", Toast.LENGTH_LONG).show();
-        return super.onCreateView(inflater, container, savedInstanceState);
+        mRootView = inflater.inflate(R.layout.fragment_topic, container, false);
+        initUiComponents();
+        return mRootView;
+    }
+
+    private void initUiComponents() {
+        ArrayList<Category> topicList = new ArrayList<>();
+        topicList.add(new Category(R.drawable.avatar_placeholder, "Business"));
+        topicList.add(new Category(R.drawable.avatar_placeholder, "Entertainment"));
+        topicList.add(new Category(R.drawable.avatar_placeholder, "Health"));
+        topicList.add(new Category(R.drawable.avatar_placeholder, "Science"));
+        topicList.add(new Category(R.drawable.avatar_placeholder, "Sport"));
+        topicList.add(new Category(R.drawable.avatar_placeholder, "Technology"));
+
+        mRecyclerView = mRootView.findViewById(R.id.categoryRecyclerView);
+        mRecyclerView.setHasFixedSize(true);
+        mLayoutManager = new LinearLayoutManager(this.getActivity());
```

Comment:
không cần this

Reply:
cho do t bi ket 1 luc, vi luc theo tut t viet bi loi vi LinearLayoutManger(this) thoi, xong ms nhan ra context phai la activity trong khi this la fragment